scripts/python/packages/parthenon_tools/parthenon_tools/movie2d.py

```diff
@@ -208,17 +208,20 @@ def plot_dump(
         ye = yf
 
     # get tensor components
-    if len(q.shape) > 6:
-        raise ValueError(
-            "Tensor rank is higher than I can handle. "
-            + "Please revise the movie2d script"
-        )
-    if len(q.shape) == 6:
-        q = q[:, components[0], components[1], 0, :, :]
-    if len(q.shape) == 5:
-        q = q[:, components[-1], 0, :, :]
-    if len(q.shape) == 4:
-        q = q[:, 0, :, :]
+    ntensors = len(q.shape[1:-3])
+    if components:
+        if len(components) != ntensors:
+            print("value error!", len(components), ntensors, q.shape)
+            raise ValueError(
+                "Tensor rank not the same as number of specified components"
```

Comment:
The exceptions raised within a thread (since the code is thread-parallel) were not being propagated, so the script was just failing silently. This was my hacky way of dealing with it.

I just went back in and fixed it properly, by forcing python to raise exceptions.